utils/generation-number.hh

```diff
@@ -8,8 +8,12 @@
 
 #pragma once
 
+#include "utils/tagged_integral.hh"
+
 namespace utils {
 
-int get_generation_number();
+using generation_type = tagged_integral<struct generation_type_tag, int>;
+
+generation_type get_generation_number();
```

Comment:
I see this wasn't added in this PR but I find it weird that we have a shared `generation_type`, it kind-of goes against the entire purpose of strong types.

Reply:
I don't like it either.

Although `get_generation_number` is generic, we effectively do use it only for gossip, so maybe it should be moved to gms and return gms::generation_type.

Reply:
I'm moving utils::get_generation_number to gms in the next version